lib/p2p/packets/Packet.ts

```diff
@@ -1,11 +1,36 @@
 import PacketType from './PacketType';
+import CryptoJS from 'crypto-js';
+import SHA256 from 'crypto-js/sha256';
+
+type PacketHeader = {
+  hash?: string;
```

Comment:
On a separate note, should this really be nullable? I figure we'd always want to and be able to determine the hash of a packet.

Reply:
I think it might make sense to make `hash` non-nullable. We could have two constructors for `Packet`, one that accepts a header which we use when deserializing a packet that we've received. The other only takes the body and an optional `reqHash` parameter, and calculates the hash of the new packet so we can set it's value right when we create the packet. Let me know what you think.

Reply:
Makes sense to try to separate, specially since TS supports constructor overloading (based on number of arguments). Will do it tomorrow.